DSharpPlus.CommandsNext/EventArgs/CommandContext.cs

```diff
@@ -92,17 +92,20 @@ internal CommandContext()
         {
             this._lazyAssMember = new Lazy<DiscordMember>(() => this.Guild != null && this.Guild.Members.TryGetValue(this.User.Id, out var member) ? member : this.Guild?.GetMemberAsync(this.User.Id).ConfigureAwait(false).GetAwaiter().GetResult());
         }
-
+        
+        // TODO Add message_id to remaining classes, but we're just testing it right now, so RespondAsync is fine.
         /// <summary>
         /// Quickly respond to the message that triggered the command.
         /// </summary>
         /// <param name="content">Message to respond with.</param>
         /// <param name="isTTS">Whether the message is to be spoken aloud.</param>
         /// <param name="embed">Embed to attach.</param>
         /// <param name="mentions">A list of mentions permitted to trigger a ping.</param>
+        /// <param name="message_id">The Id of a message to reply to.</param>
         /// <returns></returns>
-        public Task<DiscordMessage> RespondAsync(string content = null, bool isTTS = false, DiscordEmbed embed = null, IEnumerable<IMention> mentions = null) 
-            => this.Message.RespondAsync(content, isTTS, embed, mentions);
+        public Task<DiscordMessage> RespondAsync(string content = null, bool isTTS = false, DiscordEmbed embed = null, 
+            IEnumerable<IMention> mentions = null, ulong? message_id = null, bool mention = false) 
```

Comment:
I run into a lot of cases where I have a message ID from somewhere else (eg. a database lookup) and don't have a full DiscordMessage object on hand. At the moment it seems the only way to *create* a DiscordMessage object is to get it through the REST client, which would make an unnecessary API call just to get a wrapped object.

Reply:
You're able to get it from the `DiscordClient`'s message cache though right? The main reason I am against just using IDs is due to consistency because most of our other methods require a full object. IDs would also still be able to be used with the `DiscordRestClient`.

Reply:
Fair enough. We'll have to rework how we handle passing complete objects to REST, but that would be out of scope for this PR.